test/Test/Database/LSMTree/Internal/Run.hs

```diff
@@ -209,3 +229,98 @@ prop_WriteAndOpen fs hbio wb =
 
       -- TODO: return a proper Property instead of using assertEqual etc.
       return (property True)
+
+-- | Writing and loading a 'WriteBuffer' gives the same in-memory
+--   representation as the original run.
+prop_WriteAndOpenWriteBuffer ::
+     FS.HasFS IO h
+  -> FS.HasBlockIO IO h
+  -> RunData KeyForIndexCompact SerialisedValue SerialisedBlob
+  -> IO Property
+prop_WriteAndOpenWriteBuffer hfs hbio rd = do
+  withTempRegistry $ \reg -> do
+    -- Serialise run data as write buffer:
+    let srd = serialiseRunData rd
+    let inPaths = WrapRunFsPaths $ simplePath 1111
+    let resolve (SerialisedValue x) (SerialisedValue y) = SerialisedValue (x <> y)
+    withRunDataAsWriteBuffer hfs resolve inPaths srd $ \wb wbb -> do
+      -- Write write buffer to disk:
+      let wbPaths = WrapRunFsPaths $ simplePath 1312
+      withSerialisedWriteBuffer hfs hbio wbPaths wb wbb $ do
+        -- Read write buffer from disk:
+        (wb', wbb') <- readWriteBuffer reg resolve hfs hbio wbPaths
+        assertEqual "k/ops" wb wb'
+        releaseRef wbb'
+    -- TODO: return a proper Property instead of using assertEqual etc.
+    return (property True)
+
+-- | Writing run data to the disk via 'writeWriteBuffer' gives the same key/ops
+--   and blob files as when written out as a run.
+prop_WriteRunEqWriteWriteBuffer ::
+     FS.HasFS IO h
+  -> FS.HasBlockIO IO h
+  -> RunData KeyForIndexCompact SerialisedValue SerialisedBlob
+  -> IO Property
+prop_WriteRunEqWriteWriteBuffer hfs hbio rd = do
+  -- Serialise run data as run:
+  let srd = serialiseRunData rd
+  let rdPaths = simplePath 1337
+  let rdKOpsFile = Paths.runKOpsPath rdPaths
+  let rdBlobFile = Paths.runBlobPath rdPaths
+  withRun hfs hbio rdPaths srd $ \_run -> do
+    -- Serialise run data as write buffer:
+    let f (SerialisedValue x) (SerialisedValue y) = SerialisedValue (x <> y)
+    let inPaths = WrapRunFsPaths $ simplePath 1111
+    withRunDataAsWriteBuffer hfs f inPaths srd $ \wb wbb -> do
+      let wbPaths = WrapRunFsPaths $ simplePath 1312
+      let wbKOpsPath = Paths.writeBufferKOpsPath wbPaths
+      let wbBlobPath = Paths.writeBufferBlobPath wbPaths
+      withSerialisedWriteBuffer hfs hbio wbPaths wb wbb $ do
+        -- Compare KOps:
+        FS.withFile hfs rdKOpsFile FS.ReadMode $ \rdKOpsHandle ->
+          FS.withFile hfs wbKOpsPath FS.ReadMode $ \wbKOpsHandle -> do
+            rdKOps <- FSL.hGetAll hfs rdKOpsHandle
+            wbKOps <- FSL.hGetAll hfs wbKOpsHandle
+            assertEqual "writtenRunKOps/writtenWriteBufferKOps" rdKOps wbKOps
+        -- Compare Blob:
+        FS.withFile hfs rdBlobFile FS.ReadMode $ \rdBlobHandle ->
+          FS.withFile hfs wbBlobPath FS.ReadMode $ \wbBlobHandle -> do
+            rdBlob <- FSL.hGetAll hfs rdBlobHandle
+            wbBlob <- FSL.hGetAll hfs wbBlobHandle
+            assertEqual "writtenRunBlob/writtenWriteBufferBlob" rdBlob wbBlob
```

Comment:
This compares only the bytes, but we can probably compare both the keyops and blobs instead/in addition. If there are byte mismatches currently, then the assertion wouldn't create readable counter examples, because it would be thousands of bytes for even a single key/value/blob triplet